src/LanguageServer/Impl/Completion/ExpressionCompletion.cs

```diff
@@ -41,23 +44,57 @@ internal static class ExpressionCompletion {
             if (!value.IsUnknown()) {
 
                 var type = value.GetPythonType();
-                if(type is IPythonClassType cls) {
+                if (type is IPythonClassType cls) {
                     return GetClassItems(cls, e, context);
                 }
 
-                var items = new List<CompletionItem>();
-                foreach (var t in type.GetMemberNames().ToArray()) {
-                    var m = type.GetMember(t);
-                    if (m is IVariable v && v.Source != VariableSource.Declaration) {
-                        continue;
-                    }
-                    items.Add(context.ItemSource.CreateCompletionItem(t, m, type));
-                }
-                return items;
+                var memberNames = type is IPythonModule module
+                    ? GetModuleMemberNames(module, context)
+                    : type.GetMemberNames();
+
+                return memberNames.Select(name => context.ItemSource.CreateCompletionItem(name, type.GetMember(name), type));
             }
             return Enumerable.Empty<CompletionItem>();
         }
 
+        private static IEnumerable<string> GetModuleMemberNames(IPythonModule module, CompletionContext context) {
+            var variables = module.Analysis?.GlobalScope?.Variables;
+            if (variables == null) {
+                return module.GetMemberNames();
+            }
+
+            var fs = context.Services.GetService<IFileSystem>();
+            var thisModuleDirectory = Path.GetDirectoryName(module.FilePath);
+            // drop imported modules and typing.
+            return variables
+                .Where(v => {
```

Comment:
If I do `torch.`, I can see the builtin types in the completion. Should those be filtered? (It's also a little inconsistent, `torch.nn.` doesn't give `int`.)

Reply:
They appear to be of a different type, so it is not filtering issue. Pandas don't show them either.

![image](https://user-images.githubusercontent.com/12820357/65346722-37d62500-db92-11e9-8751-bca847caa65e.png)

Reply:
Ah, I see now, they're `dtype`. Wonder why you can't F12 to them...

Reply:
https://pytorch.org/docs/stable/type_info.html

Reply:
Most probably compiled. In-memory type came from stub which we don't show. `lib\site-packages\torch\__init__.pyi`